osu.Server.Queues.ScoreStatisticsProcessor/Processors/ScorePerformanceProcessor.cs

```diff
@@ -206,18 +174,29 @@ public static bool AllModsValidForPerformance(SoloScoreInfo score, Mod[] mods)
                     case ModHidden:
                     case ModFlashlight:
                     case ModMuted:
+                    case ModNightcore:
+                    case ModDoubleTime:
+                    case ModDaycore:
+                    case ModHalfTime:
                     // osu!-specific:
                     case OsuModSpunOut:
                     case OsuModTouchDevice:
                     // mania-specific:
                     case ManiaKeyMod:
                     case ManiaModMirror:
-                        continue;
+                        break;
+
+                    // Classic mod is only allowed on legacy scores.
+                    case ModClassic:
+                        return score.IsLegacyScore;
```

Comment:
This is perhaps a side note and not relevant to the core of this pull but I realise now - after making a mistake in my attempt at this - that this is _kinda_ wrong (although not in a way that would _functionally_ matter) and, if wanting to be 100% correct, should probably be

```suggestion
                    case ModClassic:
                        if (!score.IsLegacyScore)
                            return false;
                            
                        break;
```

because this is inside a switch *inside* a foreach. So what this ends up doing is that if a score is passed in here with classic mod and DT, for example, it won't bother to check the DT mod once it encounters the classic mod and checks that the score is legacy.

Maybe this was by design, I dunno.

Reply:
Good catch. I don't think this matters in terms of legacy scores, but good to have.